src/TableFunctions/TableFunctionAzureBlobStorage.cpp

```diff
@@ -44,10 +44,8 @@ bool isConnectionString(const std::string & candidate)
 
 }
 
-StorageAzureBlob::Configuration TableFunctionAzureBlobStorage::parseArgumentsImpl(ASTs & engine_args, const ContextPtr & local_context, bool get_format_from_file)
+void TableFunctionAzureBlobStorage::parseArgumentsImpl(ASTs & engine_args, const ContextPtr & local_context)
```

Comment:
I just noticed that we have `structure` argument after `compression` argument in `azureBlobStorage` function. It's quite inconvinient, in other table functions `file/s3/url/hdfs` we have the opposite order, it can lead to confusions. And in general `compression` argument is used much rarer than `structure` argument, so client will need to use `auto` for  `compression` when he needs to specify structure argument, that is also not convinient.
Was there a reason for it?
BTW, I guess it's too late to change it(

Reply:
Ah, I see we support `azureBlobStorage(url, container_name, blobpath, structure)`. Ok. We need to add a comment somewhere with all signatures we support, because `connection_string|storage_account_url, container_name, blobpath, [account_name, account_key, format, compression, structure]` is not informative at all

Reply:
And why for example we don't support `azureBlobStorage(url, container_name, blobpath, account_name, account_key, structure)`

Reply:
> I just noticed that we have structure argument after compression argument in azureBlobStorage function. It's quite inconvinient, in other table functions file/s3/url/hdfs we have the opposite order, it can lead to confusions. And in general compression argument is used much rarer than structure argument, so client will need to use auto for compression when he needs to specify structure, that is also inconvinient.
> Was there a reason for it?
> BTW, I guess it's too late to change it (or maybe we can ask in Alexandr/Alexey what they think about this backward incompatible change)

> Was there a reason for it?
No specific reason, it was implemented differently by Sasha & me.
After initial implementation of azureBlobStroage was merged, I had discussed this with @alesapin and he suggested we keep this format also because mostly in practise these parameters are set to default or not specified so it shouldn't be a problem. I understand remembering the different order for different functions is not convenient. Named collection really helps with this.